File: 0474-Ones and Zeroes/0474-Ones and Zeroes.cpp
class Solution {
public:
    int findMaxForm(vector<string>& strs, int m, int n) {
        vector<vector<int>> dp(1 + m, vector<int>(1 + n));
        for (int i = 0; i < strs.size(); ++i) {
            int count0 = 0, count1 = 0;
            for (char c : strs[i]) {
                if (c == '0') {
                    ++count0;
                }
                else {
                    ++count1;
                }
            }
            for (int j = m; j >= count0; --j) {
                for (int k = n; k >= count1; --k) {
                    dp[j][k] = max(dp[j][k], 1 + dp[j - count0][k - count1]);
                }
            }
        }
        
        return dp[m][n];
    }
};

File: 0401-0500/0474-Ones and Zeroes/0474-Ones and Zeroes.py
class Solution(object):
    def findMaxForm(self, strs, m, n):
        """
        :type strs: List[str]
        :type m: int
        :type n: int
        :rtype: int
        """
        dp = [[0]*(1 + n) for _ in range(1 + m)]
        for s in strs:
            count0 = count1 = 0
            for c in s:
                if c == '0':
                    count0 += 1
                else:
                    count1 += 1
            for i in range(m, count0 - 1, -1):
                for j in range(n, count1 - 1, -1):
                    dp[i][j] = max(dp[i][j], 1 + dp[i - count0][j - count1])
        return dp[m][n]
